### ISSUE TITLE
Narrow completion suggestions to __all__ value

### ISSUE THREAD

**Issue by m-novikov** [microsoft/pyright#3241]
**Is your feature request related to a problem? Please describe.**

Some modules import a lot of symbols into their namespace. 
Auto-completion on such modules lists own symbols as well as all imported ones, which can be inconvenient.

Let's say I have a module `a.py` containing following
```python
from typing import List

__all__ = ["foo"]

def foo() -> List[str]:
    return ["a"]
```
Currently autocomplete on `a.` will give me `a.List` and `a.foo` as possible options.

**Describe the solution you'd like**

When triggering auto complete on such module I would like to see only names listed in `__all__` in suggestions + common module attributes such as `__file__`, `__doc__`

**Additional context**
For modules that don't specify `__all__` all names should be listed as before.

**Comment by erictraut** [comment]
I don't think this matches the expectations of most uses. The `__all__` symbol is not meant to be a comprehensive list of exported symbols. Rather, it's a list of symbols that are included in a wildcard import.

Since this isn't a core type checking issue, I'm going to transfer it to the pylance-release project and let the pylance team decide how to proceed.
